internal/scanner/scanner.go

```diff
@@ -0,0 +1,145 @@
+package scanner
+
+import (
+	"io"
+	"time"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/decimal"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/types"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/value"
+)
+
+// RawValue scanning non-primitive yql types or for own implementation scanner native API
+type RawValue interface {
+	Path() string
+	WritePathTo(w io.Writer) (n int64, err error)
+	Type() types.Type
+	Bool() (v bool)
+	Int8() (v int8)
+	Uint8() (v uint8)
+	Int16() (v int16)
+	Uint16() (v uint16)
+	Int32() (v int32)
+	Uint32() (v uint32)
+	Int64() (v int64)
+	Uint64() (v uint64)
+	Float() (v float32)
+	Double() (v float64)
+	Date() (v time.Time)
+	Datetime() (v time.Time)
+	Timestamp() (v time.Time)
+	Interval() (v time.Duration)
+	TzDate() (v time.Time)
+	TzDatetime() (v time.Time)
+	TzTimestamp() (v time.Time)
+	String() (v []byte)
+	UTF8() (v string)
+	YSON() (v []byte)
+	JSON() (v []byte)
+	UUID() (v [16]byte)
+	JSONDocument() (v []byte)
+	DyNumber() (v string)
+	Value() value.Value
+
+	// Any returns any primitive or optional value.
+	// Currently, it may return one of these types:
+	//
+	//   bool
+	//   int8
+	//   uint8
+	//   int16
+	//   uint16
+	//   int32
+	//   uint32
+	//   int64
+	//   uint64
+	//   float32
+	//   float64
+	//   []byte
+	//   string
+	//   [16]byte
+	//
+	Any() interface{}
+
+	// Unwrap unwraps current item under scan interpreting it as Optional<Type> types.
+	Unwrap()
+	AssertType(t types.Type) bool
+	IsNull() bool
+	IsOptional() bool
+
+	// ListIn interprets current item under scan as a ydb's list.
+	// It returns the size of the nested items.
```

Comment:
What unit of size - bytes?

May be it return len of the list or count of items?